src/test/groovy/com/vacation_bot/vacation/VacationIntegrationTest.groovy

```diff
@@ -0,0 +1,160 @@
+package com.vacation_bot.vacation
+
+import com.vacation_bot.AbstractSpockIntegrationTest
+import com.vacation_bot.domain.models.UserModel
+import com.vacation_bot.domain.models.VacationRequestBody
+import com.vacation_bot.domain.models.VacationTotalModel
+import org.springframework.boot.test.web.client.TestRestTemplate
+import org.springframework.http.HttpEntity
+import org.springframework.http.HttpHeaders
+import org.springframework.http.HttpMethod
+import org.springframework.http.MediaType
+import org.springframework.http.ResponseEntity
+import spock.lang.Shared
+import spock.lang.Unroll
+
+import java.time.Instant
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.time.ZoneOffset
+import java.time.temporal.ChronoUnit
+
+/**
+ * Integration level test of {@Link VacationGateway}.
+ */
+class VacationIntegrationTest extends AbstractSpockIntegrationTest {
+
+    def restTemplate = new TestRestTemplate()
+
+    def headers = new HttpHeaders(accept: Arrays.asList(MediaType.APPLICATION_JSON))
+
+    String userId = UUID.randomUUID()
+    def userName = 'Alex'
+    def aliases = ['Batman', 'Joker', 'Superman']
+    @Shared
+    def startDate = '2017-10-02'
+    @Shared
+    def endDate = '2017-10-26'
+
+    @Shared
+    def parsedStartDate = LocalDate.parse(startDate)
+    @Shared
+    def parsedEndDate = LocalDate.parse(endDate)
+    @Shared
+    String period = ChronoUnit.DAYS.between(parsedStartDate, parsedEndDate)
+
+    def validUser = new UserModel(id: userId, name: userName, aliases: aliases)
+    def inputData = new VacationRequestBody(userName: userName, startDate: startDate, endDate: endDate)
+
+    @Shared
+    def static final defaultVacationTotalDays = 20
+
+    def vacationTotalDays = 18
+    def vacationTotalYear = 2017
+    def validVacationTotal = new VacationTotalModel(userId: userId, vacationTotal: vacationTotalDays, year: vacationTotalYear)
+
+    def 'exercise controller (vacation total is null)'() {
+
+        given: 'the user is saved to the db'
+        userRepository.save(validUser)
+
+        and: 'set new end date'
+        inputData.endDate = newEndDate
+
+        when: 'the controller is called'
+        String result = getResponse(inputData).getBody()
+
+        then: 'the result is correct'
+        result == expectedResult
+        def createdVacationTotal = vacationTotalRepository.findAll().stream().filter { e -> e.userId == userId && e.year == vacationTotalYear }.findFirst().get()
+        createdVacationTotal
+        createdVacationTotal.vacationTotal == expectedVacationTotalDays
+
+        def createdVacation = vacationModelRepository.findAll().stream().filter { e -> e.userId == userId }.findFirst()
+        def createdVacationPresent = createdVacation.isPresent()
+        createdVacationPresent == expectedCreatedVacation
```

Comment:
This line just check that vacation document was created. It's not enough. You should check vacation document information:
```
def createdVacations = vacationModelRepository.findAll()
1 == createdVacations.size()
with ( createdVacations.first() ) {
    userId == expUserId
    days == expectedReservedDays
    startDate == expectedStartDate
    endDate == expectedEndDate
}
```

You should understand that even if you create a vacation document with random information, this test will still pass. Try to make it more useful.